cfpb/cfgov-refresh
# Set up new browser tests that use localhost

This PR is an extension of https://github.com/cfpb/cfgov-refresh/pull/4407.

Here's what we've done:

1. Copied the current tests in `test/browser_tests/cucumber/` to `test/browser_tests/cucumber-new/` (hence all the new files below).
1. Modified the new tests to use localhost:8000 instead of tox.
1. Added some new Regs3K tests.
1. Created a `gulp test:acceptance:new` task for running the new browser tests. Specific specs can be run with `gulp test:acceptance:new --specs=regulations-search.feature`.

Travis will continue running the legacy tests via `gulp test:acceptance`. Once we [set up Jenkins to build public GitHub PRs](https://github.com/cfpb/cfgov-refresh/pull/4407#issuecomment-411284282), we'll delete the legacy tests.

## Testing

1. Pull down branch
1. Get your local version of cf.gov running at http://localhost:8000
1. Ensure that the following env vars are set (they should already be in your [`.env`](https://github.com/cfpb/cfgov-refresh/blob/master/.env_SAMPLE) file):
   - `export TEST_HTTP_HOST=localhost`
   - `export DJANGO_HTTP_PORT=8000`
1. Run `gulp test:acceptance:new`

## Notes

- Removing the ability to test with `tox` necessitates taking that step out of Travis, which has the side benefit of saving several minutes per Travis build. [This is planned for a future PR once we have another way to automate browser tests without tox, such as with Jenkins against a dev or staging server]

## Checklist
